src/external/governance/GovernanceContract.sol

```diff
@@ -0,0 +1,339 @@
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function triggerUpgradeBeaconWithTimelock(address beacon)
+        external
+        onlyCommunityOrTeamMultisig
+        accessableBeacon(beacon)
+        upgradeProcessAlreadyStarted(beacon)
+        timelockPeriodExceeded(beacon)
+    {
+        //set timelock to inaktive
+        beaconTimelock[beacon].timelockActive = false;
+
+        //Upgrade beacon
+        IInverterBeacon(beacon).upgradeTo(
+            beaconTimelock[beacon].intendedImplementation,
+            beaconTimelock[beacon].intendedMinorVersion,
+            false //this is not intended to override a shutdown
+        );
+
+        emit BeaconUpgraded(
+            beacon,
+            beaconTimelock[beacon].intendedImplementation,
+            beaconTimelock[beacon].intendedMinorVersion
+        );
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function cancelUpgrade(address beacon)
+        external
+        onlyCommunityOrTeamMultisig
+        upgradeProcessAlreadyStarted(beacon)
+    {
+        beaconTimelock[beacon].timelockActive = false;
+        emit BeaconUpgradedCanceled(beacon);
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function setTimelockPeriod(uint newTimelockPeriod)
+        external
+        onlyRole(COMMUNITY_MULTISIG_ROLE)
+        validAmount(newTimelockPeriod)
+    {
+        timelockPeriod = newTimelockPeriod;
+        emit TimelockPeriodSet(newTimelockPeriod);
+    }
+
+    //---------------------------
+    //Emergency Shutdown
+
+    /// @inheritdoc IGovernanceContract
+    function initiateBeaconShutdown(address beacon)
+        external
+        onlyCommunityOrTeamMultisig
+        accessableBeacon(beacon)
+    {
+        IInverterBeacon(beacon).shutDownImplementation();
+        emit BeaconShutdownInitiated(beacon);
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function forceUpgradeBeaconAndRestartImplementation(
+        address beacon,
+        address newImplementation,
+        uint newMinorVersion
+    )
+        external
+        onlyRole(COMMUNITY_MULTISIG_ROLE)
+        accessableBeacon(beacon)
+        validAddress(newImplementation)
+    {
+        IInverterBeacon(beacon).upgradeTo(
+            newImplementation, newMinorVersion, true
+        );
+        emit BeaconForcefullyUpgradedAndImplementationRestarted(
+            beacon, newImplementation, newMinorVersion
+        );
+    }
+
+    /// @inheritdoc IGovernanceContract
+    function restartBeaconImplementation(address beacon)
+        external
+        onlyRole(COMMUNITY_MULTISIG_ROLE)
+        accessableBeacon(beacon)
+    {
+        IInverterBeacon(beacon).restartImplementation();
+        emit BeaconImplementationRestarted(beacon);
+    }
+
+    //---------------------------
+    //Ownable2Step
+
+    /// @inheritdoc IGovernanceContract
+    function acceptOwnership(address adr)
+        external
+        onlyCommunityOrTeamMultisig //@note correct Role?
+            //@note should be beacon?
```

Comment:
I think we can keep it open to any contract, not just beacons, that we may claim ownership of.